plugin/evm/limitorders/memory_database.go

```diff
@@ -125,8 +126,11 @@ func (db *InMemoryDatabase) Delete(id string) {
 	deleteOrder(db, id)
 }
 
-func (db *InMemoryDatabase) UpdateFilledBaseAssetQuantity(quantity *big.Int, order Order) {
-	limitOrder := db.OrderMap[getIdFromOrder(order)]
+func (db *InMemoryDatabase) UpdateFilledBaseAssetQuantity(quantity *big.Int, orderId string) {
+	limitOrder := db.OrderMap[orderId]
+	if limitOrder == nil {
```

Comment:
this should never happen ideally

Reply:
I understand it wont happen but codewise i think it is better to have these instead of crashing the whole process.